In [9]:
!pip install kagglehub

In [11]:
import kagglehub
import pandas as pd

In [13]:
# Download latest version
path = kagglehub.dataset_download("yagunnersya/fifa-21-messy-raw-dataset-for-cleaning-exploring")

100%|██████████| 5.28M/5.28M [00:02<00:00, 1.90MB/s]

Extracting files...


In [15]:
print("Path to dataset files:", path)

Path to dataset files: C:\Users\steve\.cache\kagglehub\datasets\yagunnersya\fifa-21-messy-raw-dataset-for-cleaning-exploring\versions\2


In [17]:
# Read the dataset from path
df = pd.read_csv(f"{path}/fifa21 raw data v2.csv")

C:\Users\steve\AppData\Local\Temp\ipykernel_30276\1496996970.py:2: DtypeWarning: Columns (76) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(f"{path}/fifa21 raw data v2.csv")


In [19]:
df

ID               Name                      LongName  \
0      158023           L. Messi                  Lionel Messi   
1       20801  Cristiano Ronaldo  C. Ronaldo dos Santos Aveiro   
2      200389           J. Oblak                     Jan Oblak   
3      192985       K. De Bruyne               Kevin De Bruyne   
4      190871          Neymar Jr    Neymar da Silva Santos Jr.   
...       ...                ...                           ...   
18974  247223             Xia Ao                        Ao Xia   
18975  258760           B. Hough                     Ben Hough   
18976  252757        R. McKinley                Ronan McKinley   
18977  243790       Wang Zhen'ao                  Zhen'ao Wang   
18978  252520          Zhou Xiao                     Xiao Zhou   

                                               photoUrl  \
0      https://cdn.sofifa.com/players/158/023/21_60.png   
1      https://cdn.sofifa.com/players/020/801/21_60.png   
2      https://cdn.sofifa.com/players/200/389/21_60.png   
3      https://cdn.sofifa.com/players/192/985/21_60.png   
4      https://cdn.sofifa.com/players/190/871/21_60.png   
...                                                 ...   
18974  https://cdn.sofifa.com/players/247/223/21_60.png   
18975  https://cdn.sofifa.com/players/258/760/21_60.png   
18976  https://cdn.sofifa.com/players/252/757/21_60.png   
18977  https://cdn.sofifa.com/players/243/790/21_60.png   
18978  https://cdn.sofifa.com/players/252/520/21_60.png   

                                               playerUrl Nationality  Age  \
0      http://sofifa.com/player/158023/lionel-messi/2...   Argentina   33   
1      http://sofifa.com/player/20801/c-ronaldo-dos-s...    Portugal   35   
2      http://sofifa.com/player/200389/jan-oblak/210006/    Slovenia   27   
3      http://sofifa.com/player/192985/kevin-de-bruyn...     Belgium   29   
4      http://sofifa.com/player/190871/neymar-da-silv...      Brazil   28   
...                                                  ...         ...  ...   
18974     http://sofifa.com/player/247223/ao-xia/210006/    China PR   21   
18975  http://sofifa.com/player/258760/ben-hough/210006/     England   17   
18976  http://sofifa.com/player/252757/ronan-mckinley...     England   18   
18977  http://sofifa.com/player/243790/zhenao-wang/21...    China PR   20   
18978  http://sofifa.com/player/252520/xiao-zhou/210006/    China PR   21   

       ↓OVA  POT                         Club  ...     A/W     D/W   IR PAC  \
0        93   93         \n\n\n\nFC Barcelona  ...  Medium     Low  5 ★  85   
1        92   92             \n\n\n\nJuventus  ...    High     Low  5 ★  89   
2        91   93      \n\n\n\nAtlético Madrid  ...  Medium  Medium  3 ★  87   
3        91   91      \n\n\n\nManchester City  ...    High    High  4 ★  76   
4        91   91  \n\n\n\nParis Saint-Germain  ...    High  Medium  5 ★  91   
...     ...  ...                          ...  ...     ...     ...  ...  ..   
18974    47   55           \n\n\n\nWuhan Zall  ...  Medium  Medium  1 ★  64   
18975    47   67      \n\n\n\nOldham Athletic  ...  Medium  Medium  1 ★  64   
18976    47   65           \n\n\n\nDerry City  ...  Medium  Medium  1 ★  63   
18977    47   57     \n\n\n\nDalian YiFang FC  ...  Medium  Medium  1 ★  58   
18978    47   57     \n\n\n\nDalian YiFang FC  ...  Medium  Medium  1 ★  62   

      SHO  PAS DRI DEF PHY Hits  
0      92   91  95  38  65  771  
1      93   81  89  35  77  562  
2      92   78  90  52  90  150  
3      86   93  88  64  78  207  
4      85   86  94  36  59  595  
...    ..  ...  ..  ..  ..  ...  
18974  28   26  38  48  51  NaN  
18975  40   48  49  35  45  NaN  
18976  39   44  46  40  53  NaN  
18977  49   41  49  30  44  NaN  
18978  22   39  42  45  55  NaN  

[18979 rows x 77 columns]

18979 rows and 77 columns! Let's clean it!<br>
And let's first drop duplicates and check for null values

In [85]:
# df = df.drop_duplicates()
#sorted(df.isna().sum(), reverse=True)
missing_values = df.isna().sum().nlargest(5).sort_values(ascending=False).reset_index()
missing_values.columns = ['Column', 'Missing values']
missing_values

Column  Missing values
0  Loan Date End           17966
1           Hits            2595
2             ID               0
3           Name               0
4       LongName               0

We're eventually gonna have to do something about these 2595 missing values in the Hits column and decide on what to do with the Loan Date End column. <br>
Let's continue with looking at all the columns to see if there are any that we can straight up remove

In [91]:
df.columns

Index(['ID', 'Name', 'LongName', 'photoUrl', 'playerUrl', 'Nationality', 'Age',
       '↓OVA', 'POT', 'Club', 'Contract', 'Positions', 'Height', 'Weight',
       'Preferred Foot', 'BOV', 'Best Position', 'Joined', 'Loan Date End',
       'Value', 'Wage', 'Release Clause', 'Attacking', 'Crossing', 'Finishing',
       'Heading Accuracy', 'Short Passing', 'Volleys', 'Skill', 'Dribbling',
       'Curve', 'FK Accuracy', 'Long Passing', 'Ball Control', 'Movement',
       'Acceleration', 'Sprint Speed', 'Agility', 'Reactions', 'Balance',
       'Power', 'Shot Power', 'Jumping', 'Stamina', 'Strength', 'Long Shots',
       'Mentality', 'Aggression', 'Interceptions', 'Positioning', 'Vision',
       'Penalties', 'Composure', 'Defending', 'Marking', 'Standing Tackle',
       'Sliding Tackle', 'Goalkeeping', 'GK Diving', 'GK Handling',
       'GK Kicking', 'GK Positioning', 'GK Reflexes', 'Total Stats',
       'Base Stats', 'W/F', 'SM', 'A/W', 'D/W', 'IR', 'PAC', 'SHO', 'PAS',
       'DRI', 'DEF', 

In [93]:
df['BOV']

0        93
1        92
2        91
3        91
4        91
         ..
18974    49
18975    51
18976    49
18977    48
18978    50
Name: BOV, Length: 18979, dtype: int64

It seems that we will use all of these columns! The only one we might get rid of is the Hits column